src/main/java/dev/waterdog/player/HandshakeUtils.java

```diff
@@ -130,16 +130,19 @@ public static HandshakeEntry processHandshake(BedrockSession session, LoginPacke
         ECPublicKey identityPublicKey = EncryptionUtils.generateKey(identityPublicKeyString);
         JWSObject clientJwt = JWSObject.parse(packet.getSkinData().toString());
         EncryptionUtils.verifyJwt(clientJwt, identityPublicKey);
-        JsonObject clientData = HandshakeUtils.parseClientData(clientJwt, session);
+        JsonObject clientData = HandshakeUtils.parseClientData(clientJwt, extraData, session);
         return new HandshakeEntry(identityPublicKey, clientData, extraData, xboxAuth, protocol);
     }
 
-    public static JsonObject parseClientData(JWSObject clientJwt, BedrockSession session) throws Exception {
+    public static JsonObject parseClientData(JWSObject clientJwt,JsonObject extraData, BedrockSession session) throws Exception {
         JsonObject clientData = (JsonObject) JsonParser.parseString(clientJwt.getPayload().toString());
         ProxyConfig config = ProxyServer.getInstance().getConfiguration();
-        if (config.useLoginExtras() && config.isIpForward()) {
+        if (config.useLoginExtras()) {
             // Add waterdog attributes
+            clientData.addProperty("Waterdog_XUID", extraData.get("XUID").getAsString());
+            if(config.isIpForward()){
```

Comment:
I think this option might be actually removed. Would be nice if you can do it.

Reply:
what exactly do you want to delete

Reply:
` if(config.isIpForward()){`